mobility_games/games/grid_game.py

```diff
@@ -0,0 +1,123 @@
+import sys
+from rospkg import RosPack
+
+class Turns(object):
+
+	def __init__(self):
+		rospy.init_node("grid_game")
+		rospy.Subscriber("/tango_pose", PoseStamped,self.process_pose)
+
+		top = RosPack().get_path('mobility_games')
+		self.sound_folder = path.join(top, 'auditory/sound_files')
+		self.engine = pyttsx.init()
+		self.dingNoise = pw.Wav(path.join(self.sound_folder, "ding.wav"))
+		self.jumpNoise = pw.Wav(path.join(self.sound_folder, "jomp.wav"))
+		self.beepNoise = pw.Wav(path.join(self.sound_folder, "beep3.wav"))
+		self.lastDingNoise = rospy.Time.now()
+		self.lastJumpNoise = rospy.Time.now()
+		self.lastBeepNoise = rospy.Time.now()
+		self.hasSpoken = False
+
+	def process_pose(self, msg):
+		self.x = msg.pose.position.x
+		self.y = msg.pose.position.y
+		self.z = msg.pose.position.z
+		angles = euler_from_quaternion([msg.pose.orientation.x,
+										msg.pose.orientation.y,
+										msg.pose.orientation.z,
+										msg.pose.orientation.w])
+		self.yaw = angles[2]
+
+	def turn_game(self,angle):
+		self.yaw = None  
+		sound = {90:"Turn Left", -90:"Turn right", 180:"Turn around"}
+		speech = sound[angle]  
+		self.engine.say(speech)  #Says the turn accordng to the angle
+		if not self.hasSpoken:   #Make sures that it doens't get interrupted
+			self.engine.runAndWait()
+			self.engine.say(speech)
+			self.engine.runAndWait()
+		self.hasSpoken = True
+		r = rospy.Rate(10)
+		while not self.yaw:  #Let us if data is coming in
+			print self.yaw	
+			r.sleep()
+			pass
+		goal_angle = self.yaw + angle*math.pi/180
+		goal_angle %= 2*math.pi
+		threshold = 20*math.pi/180
+		while not rospy.is_shutdown():
+			if abs(angle_diff(self.yaw, goal_angle)) <= threshold: #Make sure if you hit the threshhold
+				self.jumpNoise.play()
+				self.lastJumpNoise = rospy.Time.now()
+				break  #Break from the loop
+			r.sleep()
+
+	def straight(self):
+		self.x = None
+		self.y = None
+		self.yaw = None
+		r = rospy.Rate(10)
+		while not self.yaw:  #Check to see if data is coming in
+			print "NONE"
+			r.sleep()
+			pass
+		while not self.x:  #Check to see if data is coming in
+			print "None"
+			r.sleep()
+			pass
+		straightAngle = self.yaw
+		errorAngle = 15*math.pi/180
+		current_x = self.x
+		current_y = self.y
+		while not rospy.is_shutdown():
+			if abs(angle_diff(self.yaw, straightAngle)) < errorAngle:  #Makes sure that you are walking in a straigt line
+				if rospy.Time.now() - self.lastBeepNoise > rospy.Duration(1): #Plays the beep sound every second
+					self.beepNoise.close()
+					self.beepNoise = pw.Wav(path.join(self.sound_folder, "beep3.wav"))
+					self.beepNoise.play()
+					self.lastBeepNoise = rospy.Time.now()
+			x = self.x - current_x
+			y = self.y - current_y
+			dist = math.sqrt((x**2)+(y**2))
+			if dist >= 1.5:  #Check to see if you have the goal of 1.5 meter
+				self.dingNoise.play()
+				self.lastDingNoise = rospy.Time.now()
+				degree = random.choice(angleList)  #Choose a random angle from the angleList
```

Comment:
Can we make angleList not a global variable? I think you should move it from __main__ and put it as a class attribute, i.e. `self.angleList` and define it in the __init__